src/helics/core/BrokerFactory.cpp

```diff
@@ -184,7 +184,34 @@ thread which allows the destructor to fire if need be without issue*/
 
     std::shared_ptr<Broker> findBroker(const std::string& brokerName)
     {
-        return searchableBrokers.findObject(brokerName);
+        auto brk = searchableBrokers.findObject(brokerName);
+        if (brk) {
+            return brk;
+        }
+        if (brokerName.empty()) {
+            return findBroker();
+        }
+        if (brokerName.front() == '#') {
+            return findBroker(brokerName.substr(1));
+        }
+        try {
+            auto val = std::stoull(brokerName);
```

Comment:
Users should probably avoid using findBroker if their goal is to get a broker by index. A broker given a numeric name would prevent finding a broker with the matching numeric index, and a search for a broker named "123abc" or "123xyz" would both return a broker at index 123.

Reply:
I am going to change the names of the functions,  but there needed to be a way to get a broker by index from the string to handle some situations in the webserver.